controllers/crypto/usersController.js

```diff
@@ -0,0 +1,186 @@
+const logger = require('../../utils/logger')
+const userQuery = require('../../models/crypto/users')
+
+/**
+ * Fetches the data of user
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getUser = async (req, res) => {
+  try {
+    const result = await userQuery.fetchUser(req.params.id)
+
+    if (result) {
+      return res.status(200).json({
+        message: 'User returned successfully!',
+        user: result.user
+      })
+    }
+
+    return res.boom.notFound('User doesn\'t exist')
+  } catch (error) {
+    logger.error(`Error while fetching user: ${error}`)
+    return res.boom.badImplementation('Something went wrong please contact admin')
+  }
+}
+
+/**
+ * Add new user
+ *
+ * @param req {Object} - Express request object
+ * @param req.body {Object} - user object
+ * @param res {Object} - Express response object
+ */
+const addNewUser = async (req, res) => {
+  try {
+    const user = await userQuery.addUser(req)
+
+    if (user) {
+      return res.status(201).json({
+        message: 'user added successfully!',
+        userId: user.id
+      })
+    }
+
+    return res.boom.conflict('user already exists')
+  } catch (error) {
+    logger.error(`Error while creating new user: ${error}`)
+    return res.boom.badImplementation('Something went wrong please contact admin')
+  }
+}
+
+/**
+ * Update user trainsaction history
+ *
+ * @param req {Object} - Express request object
+ * @param req.body {Object} - user object
+ * @param res {Object} - Express response object
+ */
+const updateTransactionHistory = async (req, res) => {
+  try {
+    const user = await userQuery.updateTransactionHistory(req.params.id, req.body)
+
+    if (user) {
+      return res.status(200).json({
+        message: 'TransactionHistory updated successfully!',
+        userId: user.id
+      })
+    }
+
+    return res.boom.notFound('user doesnot exists')
+  } catch (error) {
+    logger.error(`Error while updating user transaction history: ${error}`)
+    return res.boom.badImplementation('Something went wrong please contact admin')
+  }
+}
+
+/**
+ * Update user shopping history
+ *
+ * @param req {Object} - Express request object
+ * @param req.body {Object} - user object
+ * @param res {Object} - Express response object
+ */
+const updateShoppingHistory = async (req, res) => {
+  try {
+    const user = await userQuery.updateShoppingHistory(req.params.id, req.body)
+
+    if (user) {
+      return res.status(200).json({
+        message: 'shoppingHistory updated successfully!',
+        userId: user.id
+      })
+    }
+
+    return res.boom.notFound('user doesnot exists')
```

Comment:
I would like to suggest a slight change here. We can change the error message to `User does not exist` or better to 
`User doesn\'t exist` as per written in other code files of this repository. 🙂

Reply:
Yes. Let's have proper casing and no typos in messages please